instat/ucrGeom.vb

```diff
@@ -965,14 +965,23 @@ Public Class ucrGeom
         'lstAllGeoms.Add(clsgeom_path)
 
 
-        'this is just a start.. needs lots of further work
-        'clsgeom_mosaic.strGeomName = "geom_mosaic"
-        ''mandatory
-        'clsgeom_mosaic.AddAesParameter("weight ", bIsMandatory:=TRUE)
-        'clsgeom_mosaic.AddAesParameter("x", bIsMandatory:=TRUE)
-        'clsgeom_mosaic.AddAesParameter("fill", bIsMandatory:=TRUE)
-        'clsgeom_mosaic.AddAesParameter("conds", bIsMandatory:=TRUE)
-        'lstAllGeoms.Add(clsgeom_mosaic)
+        clsgeom_mosaic.strGeomName = "geom_mosaic"
+        'mandatory
+        clsgeom_mosaic.AddAesParameter("x", bIsMandatory:=True, strIncludedDataTypes:={"factor"})
+        clsgeom_mosaic.AddAesParameter("fill", bIsMandatory:=True, strIncludedDataTypes:={"factor"})
+        clsgeom_mosaic.AddAesParameter("conds", strIncludedDataTypes:={"factor"})
+        clsgeom_mosaic.AddAesParameter("weight")
+
+        'adding layerParameters
+        clsgeom_mosaic.AddLayerParameter("divider", "list", Chr(34) & "vspine" & Chr(34), lstParameterStrings:={Chr(34) & "vspine" & Chr(34), Chr(34) & "hspine" & Chr(34), Chr(34) & "vbar" & Chr(34), Chr(34) & "hbar" & Chr(34)})

```

Comment:
Actually these are the divider options when using just one factor. In general you also need to add:
```
ggmosaic::mosaic(“v”)
ggmosaic::mosaic(“h”)
ggmosaic::ddecker()
```

and also make editable as there are other lists the user could choose.